graylog2-server/src/main/java/org/graylog2/indexer/retention/RetentionStrategyBindings.java

```diff
@@ -16,16 +16,43 @@
  */
 package org.graylog2.indexer.retention;
 
+import org.graylog2.configuration.ElasticsearchConfiguration;
+import org.graylog2.indexer.retention.strategies.AbstractIndexCountBasedRetentionStrategy;
 import org.graylog2.indexer.retention.strategies.ClosingRetentionStrategy;
 import org.graylog2.indexer.retention.strategies.DeletionRetentionStrategy;
 import org.graylog2.indexer.retention.strategies.NoopRetentionStrategy;
 import org.graylog2.plugin.PluginModule;
 
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
 public class RetentionStrategyBindings extends PluginModule {
+
+    private final ElasticsearchConfiguration configuration;
+
+    public RetentionStrategyBindings(ElasticsearchConfiguration elasticsearchConfiguration) {
+        this.configuration = elasticsearchConfiguration;
+    }
+
     @Override
     protected void configure() {
-        addRetentionStrategy(DeletionRetentionStrategy.class);
-        addRetentionStrategy(ClosingRetentionStrategy.class);
-        addRetentionStrategy(NoopRetentionStrategy.class);
+        List<Class<? extends AbstractIndexCountBasedRetentionStrategy>> retentionStrategies = new LinkedList<>(Arrays.asList(DeletionRetentionStrategy.class, ClosingRetentionStrategy.class, NoopRetentionStrategy.class));
+        Set<String> disabledRetentionStrategies = configuration.getDisabledRetentionStrategies();
+
+        for (String disabledStrategy : disabledRetentionStrategies) {
+            switch (disabledStrategy) {
+                case DeletionRetentionStrategy.NAME -> retentionStrategies.remove(DeletionRetentionStrategy.class);
+                case ClosingRetentionStrategy.NAME -> retentionStrategies.remove((ClosingRetentionStrategy.class));
+                case NoopRetentionStrategy.NAME -> retentionStrategies.remove((NoopRetentionStrategy.class));
```

Comment:
would be nice if we could also disable the archiving retention this way, but I guess that's not worth the effort.
Maybe add a TODO here?

Reply:
![image](https://user-images.githubusercontent.com/98284293/218090242-3cb32616-c621-4545-ae9e-f3036dcd6094.png)
It works just that the configuration page will definitly break when there are indexSet already configured to use archiving 
![image](https://user-images.githubusercontent.com/98284293/218091093-3b54dc4c-0b88-4006-bc36-1a011eecfd14.png)
@ousmaneo  maybe you can help, so that we change the page that the configuration is no longer available? There is actually a nice notification triggered https://user-images.githubusercontent.com/98284293/218089232-dddc045b-69ef-468c-b623-065e68401a3e.png

Reply:
>  It works just that the configuration page will definitly break when there are indexSet already configured to use archiving
 

isn't that true for any of those disabled retentions that are already enabled somewhere? How are we supposed to handle that in the backend? 
We could migrate them to one of the remaining retentions, but it feels a bit hackish.
I'm not sure what's a good approach here. Has @boosty given any input on this yet?

Reply:
Prior to rolling this change out to Cloud, we will check if there are unsupported options currently being used. 

I would like to discuss that part in https://github.com/Graylog2/graylog-plugin-cloud/issues/1081.